### ISSUE TITLE
Request to add Image Classification in Deep learning

### ISSUE THREAD

**Issue by BirajCoder** [Darknez07/MachineLearningModels#5]
Hey I am a beginner and I have a project on Image Classification on CIFAR 100 dataset. I would request you to allow me to add that here if possible.

**Comment by Darknez07** [comment]
@BirajCoder  let's see what do u have, Show me the code(i.e. model)

**Comment by BirajCoder** [comment]
May I share the link to my notebook here?

**Comment by Darknez07** [comment]
Yes you may @BirajCoder

**Comment by BirajCoder** [comment]
https://jovian.ml/birajde9/cifar-100 
Here is the link to my notebook , and I also have a blog in medium , 
https://medium.com/jovianml/image-classification-with-cifar100-deep-learning-using-pytorch-9d9211a696e
Please have e a look into it.

**Comment by Darknez07** [comment]
@BirajCoder  Can you tell me why the accuracy is less? 72%, and why did you use convolution with 3 kernel size, why not an alternative pipeline with 5*5 then 3*3 conv2d layer i.e. 
Conv5x5
Batch norm
Relu
Conv3X3
Batch norm
Relu
Then while getting the flat layer 
Why didn't you kill the neurons?
then the hold probability.

**Comment by BirajCoder** [comment]
As you can see from my blog I have tried a lot of other models including pretrained models but was able to achieve only 72% , but I will try the approach you said and see if the accuracy improves.

**Comment by BirajCoder** [comment]
Hey for a 5*5 convolution we will need a padding of 2 correct? I am asking as I am new in this field and I would appreciate your help .

**Comment by Darknez07** [comment]
Yea it depends on the data, padding are of two types padding or no padding, so it's on you that you want to pad or not, basically just use p = (k - 1)/2 to define the padding as "p" and "k" as kernel size. But basically it is suggested  to have it in image based dataset.

**Comment by BirajCoder** [comment]
Got it, Thanks for the help .

**Comment by BirajCoder** [comment]
Hey I tried applying conv 5 x 5 also and tried a few other ways too , I wasn't able to get past 72 % accuracy, I searched the internet and I found that as there are only 500 images per class and 100 classes it gets really hard to push accuracy in this model , still  93 % accuracy on this dataset was achieved in 2020 using Bit Transfer L pretrained model, as I am a beginner in this field I don't know properly how to use pretrained model but still I am trying to push the accuracy using EffecientNET model but I am stuck in the 70 % zone , What shall I do ?

**Comment by Darknez07** [comment]
Ohh i will have to study some more than @BirajCoder , Is there any improvements with pipelines?

**Comment by Darknez07** [comment]
@BirajCoder  here you go https://github.com/weiaicunzai/pytorch-cifar100

**Comment by BirajCoder** [comment]
I saw the repository you gave me and here too the top1 loss is between 20-40% that means my validation accuracy of 72% is somewhat at par with different models used here correct?

**Comment by Darknez07** [comment]
yes i looked into some things and i guess i have to say resnet worked good with more epoches and lr of 0.002

**Comment by BirajCoder** [comment]
So what shall I do next ? Can I upload my program here?

**Comment by Darknez07** [comment]
Upload in DL repository(make a new one) edit the readme with what have you done and basics like cnn and  maxpooling and attach your model description. and create a pull request

**Comment by BirajCoder** [comment]
Sure , I will be uploading it soon , Thanks for giving me the opportunity

**Comment by BirajCoder** [comment]
Hey I have sent a pull request , do check it and let me know .

**Comment by Darknez07** [comment]
@BirajCoder  ok

**Comment by Darknez07** [comment]
Closed with #7
